app/src/main/res/layout/fragment_deposit_key.xml

```diff
@@ -4,108 +4,104 @@
 
-            <ImageView
-                android:id="@+id/qr"
-                android:layout_width="200dp"
-                android:layout_height="200dp"
-                tools:ignore="ContentDescription"/>
+                <ImageView
+                    android:id="@+id/copy_iv"
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:layout_marginEnd="12dp"
+                    android:paddingStart="8dp"
+                    android:paddingEnd="8dp"
+                    android:paddingBottom="8dp"
+                    android:foreground="@drawable/mixin_ripple"
+                    android:src="@drawable/ic_copy_gray"
+                    app:layout_constraintTop_toTopOf="@id/key_code"
+                    app:layout_constraintEnd_toEndOf="parent"/>
 
-            <one.mixin.android.widget.BadgeCircleImageView
-                android:id="@+id/qr_avatar"
-                android:layout_width="36dp"
-                android:layout_height="36dp"
-                android:layout_gravity="center"/>
-        </FrameLayout>
-    </androidx.constraintlayout.widget.ConstraintLayout>
+                <FrameLayout
+                    android:id="@+id/qr_fl"
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:layout_alignParentEnd="true"
+                    android:layout_centerVertical="true"
+                    android:layout_marginTop="4dp"
+                    app:layout_constraintStart_toStartOf="parent"
+                    app:layout_constraintEnd_toEndOf="parent"
+                    app:layout_constraintTop_toBottomOf="@id/key_code">
 
-    <View
-        android:layout_width="match_parent"
-        android:layout_height="15dp"
-        android:background="?attr/bg_window"/>
+                    <ImageView
+                        android:id="@+id/qr"
+                        android:layout_width="240dp"
+                        android:layout_height="240dp"
+                        tools:ignore="ContentDescription" />
 
-    <TextView
-        android:id="@+id/confirm_tv"
-        android:layout_width="match_parent"
-        android:layout_height="wrap_content"
-        android:layout_marginStart="24dp"
-        android:layout_marginEnd="24dp"
-        android:paddingStart="8dp"
-        android:paddingTop="3dp"
-        android:paddingEnd="8dp"
-        android:paddingBottom="3dp"
-        android:textColor="?attr/text_minor"
-        android:textSize="13sp"/>
+                    <one.mixin.android.widget.BadgeCircleImageView
+                        android:id="@+id/qr_avatar"
+                        android:layout_width="36dp"
+                        android:layout_height="36dp"
+                        android:layout_gravity="center" />
+                </FrameLayout>
+            </androidx.constraintlayout.widget.ConstraintLayout>
+
+            <TextView
```

Comment:
No margin bottom?

Reply:
QR image has a large margin around it self, so no more margin needed.